packages/@react-spectrum/dnd/docs/dragAndDrop.mdx

```diff
@@ -0,0 +1,256 @@
+</TypeContext.Provider>
+
+`useDragHooks` will then provide you with a set of drag hooks that you can pass to the component via its `dragHooks` prop, thus enabling drag operations for the component.
+Drop operations can be enabled in the same way by passing the hooks from `useDropHooks` to the component's `dropHooks` prop.
+
+Of the various hook options above, `getAllowedDropOperations` and `getDropOperation` may be of particular interest since it allows you to specify what kinds of drag and drop operations you want to allow.
+When the dragged items are dropped on a drop target created using the React Aria drag and drop hooks, the allowed drop operations you return in `getAllowedDropOperations`
+are provided to the drop target's `getDropOperation`, giving the drop target extra information which it can use to decide what drop operation to execute. This in turn provides
+the `onDragEnd` and `onDrop` handlers with the executed drop operation, allowing you to decide what to do with the dragged items in your original collection and in the dropped collection.
+
+For instance, you may have a draggable collection of items that allows `move` and `copy` operations but you need a way to know whether or not you should be removing the dragged items
+from the list after a drop operation. A drop target that only allows `copy` operations, such as a file upload drop zone, would be able to return `copy` from its `getDropOperation` and communicate that
+to your draggable collection's `onDragEnd` handler, letting it know that it shouldn't remove the dragged items from the list. Alternatively, a drop target that allows `move` operations, like in the
+[example](#example) below, would return `move` from its `getDropOperation` and thus inform your draggable collection to remove the dragged items from it list.
+
+You may be wondering why we didn't simply import the drag and drop hooks within the components themselves. The reason behind this is to reduce the bundle size of
+the component for users who don't need the drag and drop support for their component.
+
+The following list shows which components currently support drag and drop:
+-  [ListView](ListView.html#drag-and-drop)
+
+## Example
+
+The example below illustrates how one would use `useDragHooks` and `useDropHooks` to enable dragging files from one ListView to another. For the first ListView, we want to make the rows draggable and
+have the dragged rows removed from the list upon a successful drop. To accomplish this, we first want to specify the types of data associated with each item via the array returned by `getItems`. In this case,
+we add a `text/plain` type that matches the text value associated with each item. The item's type is also included with the item's `id` mapped to it. We also provide a `onDragEnd` event handler to `useDragHooks`
+that removes the dragged items from the list only upon a successful drop operation.
+
+<details>
```

Comment:
Why are the codes hidden?

Reply:
Felt like they took up a lot of space visually between the explanation and the example itself, but will remove the <details> now that we are moving the example to the top